rust/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -320,3 +271,49 @@ where
         Ok(result)
     }
 }
+
+/// Generate a call to ChainSetup for the given builder
+macro_rules! build_store_and_retrieve {
+    ($vis:vis, $setter:ident, $getter:ident, $prefix: ident, $key: ty, $val: ty) => {
```

Comment:
nit: reformat as:
```rs
($($vis:vis)? $name_base:ident, $key_prefix: ident, $key_ty:ty, $val_ty:ty)
```

Then you can use

```rs
build_store_and_retrieve!(pub message_id_by_nonce, MESSAGE_ID, u32, H256);
```

To do this you will need to use `paste` for the store/retrieve but this will prevent needing to type it twice and also enforce the prefixes.

Further it is better to have the macro write the `impl HyperlaneRocksDB` for each since the macro is scoped outside the impl. That way the restriction to have the macro call is not within the impl block. (Yes, this means there will be multiple impl blocks generated, this is not an issue.)

Reply:
Ah nice, didn't know about paste